### ISSUE TITLE
RHS menus are promoting the first section

### ISSUE THREAD

**Issue by akimd** [docker/docker.github.io#6256]
For instance:

https://docs.docker.com/docker-for-windows/

<img width="301" alt="capture d ecran 2018-03-19 a 16 09 33" src="https://user-images.githubusercontent.com/232441/37605832-5d6460b6-2b94-11e8-904b-1cc25e7fb39e.png">

So "Check versions of Docker Engine, Compose, and Machine" is clearly higher than what follows, however the structure is:
```
     30:## Check versions of Docker Engine, Compose, and Machine
     45:## Explore the application and run examples
    242:## Set up tab completion in PowerShell
    322:## Docker Settings
    352:### General
    374:### Shared Drives
    425:#### Firewall rules for shared drives
    442:#### Shared drives on demand
    458:### Advanced
    468:### Network
    488:### Proxies
    513:### Docker daemon
    528:#### Experimental mode
    570:#### Custom registries
    583:#### Edit the daemon configuration file
    605:### Switch between Windows and Linux containers
    627:#### Getting started with Windows containers
    649:#### About the Docker Windows containers specific dialogs
    674:### Docker Store
    683:### Docker Cloud
    708:### Kubernetes
    759:### Giving feedback and getting help
    774:### Diagnose and Feedback
    799:### Reset
    813:## Adding TLS certificates
    825:## Where to go next
```

That is something I have observed on other pages (but I cannot find any other example ATM).

FWIW, this is Safari.

**Comment by gbarr01** [comment]
@akimd, you're talking about the right hand nav, correct?

**Comment by akimd** [comment]
Yup.

**Comment by joaofnfernandes** [comment]
This is a fun one! If you look at the source all entries with `##` are being generated as `H2`, so everything is correct with the markdown to HTML generation process.

What's happening here is that the heading is too long and when it starts wrapping, it can be out of alignment with the other headings. The easy (and best fix) is to change the headings to have smaller text. I think we should go with that option.

Another option is to increase the width of the right column, but I don't think we should do that. The content at the center deserves as much real estate we can give it.

**Comment by gbarr01** [comment]
@joaofnfernandes ... what about this page? It has a short heading and still does it: https://docs.docker.com/docker-for-mac/

**Comment by joaofnfernandes** [comment]
You're right. What those two pages have in common is that both of them have
```
---
toc_max: 4
toc_min: 1
---
```

So I was wondering if that could be the source of the problem. I've added those two properties to the frontmatter of a page that was rendering properly and I was able to reproduce the issue.

So the problem doesn't seem to be the long title, but the way we generate the right nav, specifically when those two properties are used.

In case anyone wants to dig a bit more, the logic for the [right nav seems to be this](https://github.com/docker/docker.github.io/blob/master/_includes/toc_pure_liquid.html).
